src/rust/src/bridge.rs

```diff
@@ -19,16 +19,35 @@ use crate::{
         new_bundle_assembler, BatchValidator, Bundle as SaplingBundle,
         BundleAssembler as SaplingBundleAssembler, Prover, Verifier,
     },
+    streams::{
+        from_auto_file, from_buffered_file, from_data, from_hash_writer, from_size_computer,
+        CppStream,
+    },
     wallet_scanner::{init_batch_scanner, BatchResult, BatchScanner},
 };
 
 #[cxx::bridge]
 pub(crate) mod ffi {
     extern "C++" {
+        include!("hash.h");
         include!("streams.h");
 
         #[cxx_name = "RustDataStream"]
         type RustStream = crate::streams::ffi::RustStream;
+        type CAutoFile = crate::streams::ffi::CAutoFile;
+        type CBufferedFile = crate::streams::ffi::CBufferedFile;
+        type CHashWriter = crate::streams::ffi::CHashWriter;
+        type CSizeComputer = crate::streams::ffi::CSizeComputer;
+    }
+    #[namespace = "stream"]
+    extern "Rust" {
+        type CppStream<'a>;
```

Comment:
Clippy lints are failing due to something inside the `cxx` macros. I cannot make this named lifetime implicit here because it is used to define a type, but it is also used somewhere inside the expanded bridge definitions below, and an `allow` annotation isn't able to reach the place where the lint triggers.